### PR TITLE
netstandard2.1 support

### PR DESCRIPTION
**Pull request by DiFFoZ**
- [ ]  Depends on #717

- Added `netstandard2.1` to target framework.
- Updated Standalone .NET version to 7.0
- Removed `Microsoft.NETFramework.ReferenceAssemblies.net461` package, it's not needed.
- Removed unused Fody weavers.
- Fixed Fody warnings about missing `PrivateAssets`.
- Fixed GitHub source link duplicate package warnings.
- Fixed builds are not `Deterministic` (required by SourceLink)

### CODE REVIEW

File: standalone/OpenMod.Standalone/OpenMod.Standalone.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net461;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net7.0</TargetFrameworks>
```
Comment (CyberAndrii) [inline standalone/OpenMod.Standalone/OpenMod.Standalone.csproj:3]: Also https://github.com/openmod/openmod/blob/main/.github/workflows/OpenMod.Standalone.yaml#L46..L71

**Review by CyberAndrii — CHANGES_REQUESTED** [review]

**Review by CyberAndrii — CHANGES_REQUESTED** [review]
Sorry, this didn't come to my mind earlier, but we still need to figure out https://github.com/openmod/openmod/issues/718 before this PR can be merged. That's because it will no longer be possible to install .NET Framework libraries if we ship OpenMod.NuGet targeting .NET Standard.

Alternatively, we could make both OpenMod.NuGet and the Module target net481 (or whatever version Unturned is going to use) like it is right now with net461.

**Comment by DiFFoZ** [comment]
What about of implementing a method for NuGet manager, e.g:
```cs
public void IncludeAvailableFrameworks(params NuGetFramework[] frameworks)
```

And then module should call it to add the range of supported frameworks.
Unturned:
```
IncludeAvailableFrameworks(net461-net481)
```
Standalone:
```
IncludeAvailableFrameworks(net6.0-net7.0)
```

**Comment by DiFFoZ** [comment]
I have got implemented it in https://github.com/openmod/openmod/pull/737/commits/1ac3bc925ca8f40dc7a759da6cf7c67bf48ad211.

If `OpenMod.NuGet` was downloaded with version `.netstandard2.1`, then it will check for if it's running under Mono, and if it does, then adds `.net4.8.0` to the supported framework.

If a version is not `.netstandard2.1` or is not running under Mono, then it will only use the target framework version of `OpenMod.Nuget`. And for that I have added `.net4.8.0` to the supported framework.

**Comment by DiFFoZ** [comment]
Updated some packages with major changes, should be not cause any breaking changes.
https://autofac.readthedocs.io/en/latest/whats-new/upgradingfrom6to7.html
https://github.com/serilog/serilog/releases/tag/v3.0.0
https://github.com/andrewlock/NetEscapades.Configuration/issues/75#issuecomment-1469506349 (minor breaking change due to using reflection)

But some of them created breaking change by pinning the version of `Microsoft.*` package to v6 or v7, we are still using v3. So these are not updated to the major version.
https://github.com/serilog/serilog-extensions-hosting/releases/tag/v7.0.0
https://github.com/autofac/Autofac.Extensions.DependencyInjection/releases/tag/v8.0.0

**Pull request merged**